jobs/basic-company/main.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"time"
+
+	"golang.org/x/oauth2/google"
+	jobs "google.golang.org/api/jobs/v2"
+)
+
+const requestDeadline = 10 * time.Second
+
+func prettyFormat(v interface{}) string {
+	j, err := json.MarshalIndent(v, "", "    ")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return string(j)
+}
+
+// [START basic_company]
+
+func makeCompany() *jobs.Company {
+	return &jobs.Company{
+		DisplayName:          "Google",
+		HqLocation:           "1600 Amphitheatre Parkway Mountain View, CA 94043",
+		DistributorCompanyId: fmt.Sprintf("company:%d", rand.New(rand.NewSource(time.Now().UnixNano())).Int63()),
+	}
+}
+
+// [END basic_company]
+
+// [START create_company]
+
+func createCompany(ctx context.Context, js *jobs.Service, toCreate *jobs.Company) (*jobs.Company, error) {
```

Comment:
A common pattern in our samples is to pass an `io.Writer`, then print to it with `fmt.Fprintf(w, ...)`. Then, tests can use a `bytes.Buffer` to verify output. For example, see https://github.com/GoogleCloudPlatform/golang-samples/blob/master/dlp/dlp_snippets/deid_test.go#L39.